Dockerfiles/cpp/.devcontainer/Dockerfile

```diff
@@ -84,29 +82,35 @@ RUN apt-get --fix-missing update && apt-get upgrade -y && \
         update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-14 100 && \
         update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-14 100
 
-# Install devcontainer features
-ADD https://raw.githubusercontent.com/devcontainers/features/7fa90110d762797cc0b1c2fe8fcc028c9b813d56/src/common-utils/install.sh /container/scripts/install-common.sh
-RUN UID="4000" bash /container/scripts/install-common.sh 
-ADD https://raw.githubusercontent.com/devcontainers/features/7fa90110d762797cc0b1c2fe8fcc028c9b813d56/src/docker-in-docker/install.sh /container/scripts/install-dind.sh
-RUN VERSION="latest" bash /container/scripts/install-dind.sh
+# Create non-root user
+ARG USERNAME=basecontainer
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+
+RUN groupadd --gid $USER_GID $USERNAME \
+    && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME \
+    && apt-get update \
+    && apt-get install -y sudo \
+    && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
+    && chmod 0440 /etc/sudoers.d/$USERNAME
 
-COPY ./common/ /scripts/
+COPY ./common /scripts
 
 ARG TARGETARCH
 WORKDIR /scripts
 RUN ./install-python.sh && \
     ./install-dapr-cli.sh && \
     ./install-k3d-prerequisites.sh && \
     ./install-velocitas-cli.sh ${TARGETARCH}
-USER 4000
+USER 1000
 RUN ./install-k3d-tooling.sh
 
-ADD --chown=4000 https://raw.githubusercontent.com/eclipse-velocitas/vehicle-app-cpp-sdk/main/requirements.txt  /container/requirements.txt
+ADD --chown=1000 https://raw.githubusercontent.com/eclipse-velocitas/vehicle-app-cpp-sdk/main/requirements.txt  /container/requirements.txt
 # Install Python requirements as vscode user and clean up
-RUN pip3 install -r /container/requirements.txt && sudo rm /container/requirements.txt
+RUN pip3 install -r /container/requirements.txt
```

Comment:
You are installing all depenedncies on pip under "basecontainer" user. It means, that in runtime on vscode, all this dependencies will be not available for user "vscode".

Reply:
A few lines below the build dependencies from this base container are copied to `vscode` users user home, so they are available